src/index.js

```diff
@@ -19,21 +26,90 @@ const renderTasks = () => {
       listItem.className = task.completed ? 'task-item completed' : 'task-item';
+      updateLocalStorage();
     });
 
     const label = document.createElement('label');
-    label.innerHTML = `<span>${task.description}</span><i class="fas fa-ellipsis-v"></i>`;
+    const descriptionSpan = document.createElement('span');
+    descriptionSpan.innerText = task.description;
+
+    const editInput = document.createElement('input');
+    editInput.setAttribute('type', 'text');
+    editInput.className = 'edit-input';
+    editInput.style.display = 'none';
+    editInput.value = task.description;
+    editInput.addEventListener('keydown', (e) => {
+      if (e.key === 'Enter') {
+        descriptionSpan.innerText = editInput.value.trim();
+        task.description = editInput.value.trim();
+        editInput.style.display = 'none';
+        descriptionSpan.style.display = 'inline';
+        updateLocalStorage();
+      }
+    });
+
+    label.appendChild(descriptionSpan);
+    label.appendChild(editInput);
+
+    label.addEventListener('click', () => {
+      descriptionSpan.style.display = 'none';
+      editInput.style.display = 'inline';
+      editInput.focus();
+    });
+
+    const deleteIcon = document.createElement('i');
+    deleteIcon.className = 'fas fa-trash delete-icon';
+    deleteIcon.style.display = 'none'; // Hide the delete icon initially
+
+    const deleteTaskHandler = (index) => {
+      tasks = deleteTask(tasks, index);
+      updateLocalStorage();
+      renderTasks(tasks);
+    };
+
+    deleteIcon.addEventListener('click', () => {
+      deleteTaskHandler(task.index);
+    });
+
+    const ellipsisIcon = document.createElement('i');
+    ellipsisIcon.className = 'fas fa-ellipsis-v ellipsis-icon';
+    ellipsisIcon.addEventListener('click', () => {
+      deleteIcon.style.display = 'inline'; // Show the delete icon
+      ellipsisIcon.style.display = 'none'; // Hide the ellipsis icon
+    });
 
     listItem.appendChild(checkbox);
     listItem.appendChild(label);
+    listItem.appendChild(ellipsisIcon);
+    listItem.appendChild(deleteIcon);
     todoList.appendChild(listItem);
   });
 };
 
-const clearCompleted = () => {
-};
+const refreshIcon = document.getElementById('refresh-icon');
+refreshIcon.addEventListener('click', () => {
+  renderTasks(tasks);
+});
 
-renderTasks();
+const addIcon = document.getElementById('add-icon');
+addIcon.addEventListener('click', () => {
+  const taskInput = document.getElementById('task-input');
+  const description = taskInput.value.trim();
+  if (description) {
+    tasks = addTask(tasks, description);
+    taskInput.value = '';
+    updateLocalStorage();
+    renderTasks(tasks);
+  }
+});
 
 const clearButton = document.getElementById('clear-button');
-clearButton.addEventListener('click', clearCompleted);
+clearButton.addEventListener('click', () => {
+  tasks = clearCompleted(tasks);
+  updateLocalStorage();
+  renderTasks(tasks);
+});
+
+renderTasks(tasks);
+
```

Comment:
- [ ] Good to have your To-Do List saved in local storage :+1: But if I add tasks and refresh the page, the added tasks are not present on the UI so please make sure you fix that. :pray: 
  
  [Screencast from 07-12-2023 03:10:39 AM.webm](https://github.com/mershark/To-do-list/assets/92303891/a30d8211-6f3d-472c-8c05-167e5fbfd0ed)